### PR TITLE
feat: Epnow control improvments (AEGHB-310)

### PR DESCRIPTION
**Pull request by n3b0j5a**
* Added battery attribute
* Added function to clear bind list
* Made bind list max size public (needed for data allocation)
* Small code style fixes

### CODE REVIEW

**Comment by lhespress** [comment]
@n3b0j5a Thanks for your contribution.

File: src/control/src/espnow_ctrl.c [hunk 1]
```diff
@@ -33,21 +33,23 @@
...
 
 static const char *TAG = "espnow_ctrl";
+
 static espnow_bindlist_t g_bindlist = {0};
 
+int8_t g_rssi;
+uint32_t g_timestamp;
+espnow_ctrl_bind_cb_t g_bind_cb;
+espnow_ctrl_data_cb_t g_data_cb;
+espnow_ctrl_data_raw_cb_t g_data_raw_cb;
```
Comment (bubblesnake) [inline src/control/src/espnow_ctrl.c:51]: Please initialize all global variables.

**Review by bubblesnake — CHANGES_REQUESTED** [review]

**Review by lhespress — COMMENTED** [review]
Is it better to rename it to espnow_ctrl_responder_load_bindlist and add it in the head file, call it by application firstly.

**Review by lhespress — COMMENTED** [review]
Is it better to rename it to espnow_ctrl_responder_load_bindlist and add it in the head file, call it by application firstly.

File: src/control/src/espnow_ctrl.c [hunk 2]
```diff
@@ -90,8 +92,18 @@ static const espnow_frame_head_t g_initiator_frame = {
 };
 #endif
 
+static void espnow_load_bindlist(void)
```
[the remark this answers is not in the document]
Comment (n3b0j5a) [reply src/control/src/espnow_ctrl.c:95]: @lhespress Naming change suggestion is fine, but i would not make necessary for application to call that function first. It should be probably called from espnow_init or something like that.

File: src/control/src/espnow_ctrl.c [hunk 3]
```diff
@@ -44,11 +44,11 @@ static const char *TAG = "espnow_ctrl";
...
-int8_t g_rssi;
-uint32_t g_timestamp;
-espnow_ctrl_bind_cb_t g_bind_cb;
-espnow_ctrl_data_cb_t g_data_cb;
-espnow_ctrl_data_raw_cb_t g_data_raw_cb;
+int8_t g_rssi = 0;
+uint32_t g_timestamp = 0;
+espnow_ctrl_bind_cb_t g_bind_cb = NULL;
+espnow_ctrl_data_cb_t g_data_cb = NULL;
+espnow_ctrl_data_raw_cb_t g_data_raw_cb = NULL;
```
Comment (nielsnl68) [inline src/control/src/espnow_ctrl.c:51]:
```cpp
espnow_ctrl_bind_cb_t g_bind_cb = nullptr;
espnow_ctrl_data_cb_t g_data_cb = nullptr;
espnow_ctrl_data_raw_cb_t g_data_raw_cb = nullptr;
```
Are you sure that **NULL** works and should it not set to **nullptr**?

Comment (bubblesnake) [reply src/control/src/espnow_ctrl.c]: Hi @nielsnl68 , nullptr is introduced in C++11. I can understand the need to migrate to nullptr in C++. Could you elaborate why we need nullptr in C here? Are you able to build with nullptr here? For me using nullptr is an undeclared error when building this component. Thanks.

**Comment by lhespress** [comment]
@n3b0j5a Could you squash these commits to one?

**Comment by LeeDr** [comment]
Are there any docs or examples on using the battery attributes?

**Pull request closed without merging**